src/planner/PlannerContext.tsx

```diff
@@ -720,73 +726,91 @@ export const usePlannerContext = (props: PlannerContextProps): PlannerContextDat
-                            const newPlan = cloneDeep(prevPlanState);
-                            const affectedInstances = newPlan.spec.blocks.filter((instance) =>
-                                parseKapetaUri(instance.block.ref).equals(blockUri)
-                            );
-
-                            affectedInstances.forEach((instance) => {
-                                newPlan.spec.connections?.filter((conn) => {
-                                    if (
-                                        role === ResourceRole.CONSUMES &&
-                                        conn.consumer.blockId === instance.id &&
-                                        conn.consumer.resourceName === resourceName
-                                    ) {
-                                        conn.consumer.resourceName = resource.metadata.name;
-                                    }
-
-                                    if (
-                                        role === ResourceRole.PROVIDES &&
-                                        conn.provider.blockId === instance.id &&
-                                        conn.provider.resourceName === resourceName
-                                    ) {
-                                        conn.provider.resourceName = resource.metadata.name;
-                                    }
-                                });
+                    if (!existingBlockInfo) {
+                        throw new Error(`Block #${blockRef} not found`);
+                    }
+
+                    const block = (blockAssets[existingBlockInfo.index] = cloneDeep(existingBlockInfo.block));
+
+                    const resources: Resource[] =
+                        role === ResourceRole.PROVIDES
+                            ? block.content.spec.providers ?? []
+                            : block.content.spec.consumers ?? [];
+
+                    const existingResourceIx = resources.findIndex((r: Resource) => r.metadata.name === resourceName);
+                    if (existingResourceIx === -1) {
+                        throw new Error(`Resource ${resourceName} on block #${blockRef} not found`);
+                    }
+
+                    const existingResource = resources[existingResourceIx];
+
+                    if (existingResource.metadata.name !== resource.metadata.name) {
+                        // We changed the name of the resource.
+                        // We need to update all connections that reference this resource
+                        newState.plan = cloneDeep(prevState.plan);
+                        const newPlan = newState.plan;
+                        const affectedInstances = newPlan.spec.blocks.filter((instance) =>
+                            parseKapetaUri(instance.block.ref).equals(blockUri)
+                        );
+                        affectedInstances.forEach((instance) => {
+                            newPlan.spec.connections?.filter((conn) => {
```

Comment:
Seems odd to use a `filter` if we don't use the return value 🤔 Should this be a forEach as well?